### PR TITLE
httpclient: Add options to hide sensitive headers values on error

### PR DESCRIPTION
**Pull request by dsabati**
A backport sur la 1.34.x

### CODE REVIEW

File: src/utils/http-client-base.ts [hunk 1]
```diff
@@ -86,8 +139,17 @@ export class N9HttpClient {
 		queryParams?: N9HttpClientQueryParams,
 		headers: object = {},
 		options: Options = {},
+		sensitiveHeadersOptions: N9HttpClientSensitiveHeadersOptions = {},
```
Comment (BenjD90) [inline src/utils/http-client-base.ts]: I don't think that should be rewritable at each call.

```suggestion
```

**Review by BenjD90 — CHANGES_REQUESTED** [review]

File: src/utils/http-client-base.ts (hunk 1 above)
Comment (dsabati) [reply src/utils/http-client-base.ts]: @BenjD90 removed!

File: src/options.ts [hunk 2]
```diff
@@ -191,6 +191,22 @@ function applyHttpOptionsDefaults(options: N9NodeRouting.Options): void {
 	}
 }
 
+function applyHttpClientOptionsDefaults(options: N9NodeRouting.Options): void {
+	// Defaults options for http client
+	const defaulSensitiveHeadersOptions: N9NodeRouting.HttpClientSensitiveHeadersOptions = {
```
Comment (BenjD90) [inline src/options.ts]: The check of env is missing to manage NODE_ENV=development

Check how it is done elsewhere.

**Review by BenjD90 — CHANGES_REQUESTED** [review]

File: src/options.ts (hunk 2 above)
Comment (dsabati) [reply src/options.ts]: @BenjD90 done!

File: src/utils/http-client-base.ts [hunk 3]
```diff
@@ -2,20 +2,30 @@ import { N9Log } from '@neo9/n9-node-log';
...
 
 export type N9HttpClientQueryParams =
 	| string
 	| Record<string, string | number | boolean | string[] | number[] | boolean[]>
 	| object;
 
+export class N9HttpClientOptions extends HttpClientOptions {
+	gotOptions?: GotOptions;
+}
+
```
Comment (BenjD90) [inline src/utils/http-client-base.ts]: To be move to the HttpClientOptions, without deep validation.

**Review by BenjD90 — CHANGES_REQUESTED** [review]

File: src/utils/http-client-base.ts (hunk 3 above)
Comment (dsabati) [reply src/utils/http-client-base.ts]: @BenjD90 done!

**Review by BenjD90 — APPROVED** [review]

**Pull request merged**